modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -176,48 +181,62 @@ public CoreEvent process(CoreEvent event) throws MuleException {
   @Override
   public Publisher<CoreEvent> apply(Publisher<CoreEvent> publisher) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to reduce its Cognitive Complexity from 21 to the 15 allowed. ([squid:S3776](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS3776))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7661&issues=AWjOP1ygotIXnmsUjewY&open=AWjOP1ygotIXnmsUjewY)

[ISSUE_KEY]: AWjOP1ygotIXnmsUjewY